### PR TITLE
Add basic telemetry to track anonymous usage

### PR DESCRIPTION
**Pull request by malcolmholmes**

### CODE REVIEW

File: src/telemetry.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        await sendEvent("first", uuid);
+        ctx.globalState.update(LAST_UPDATED_DATE, today);
+        ctx.globalState.update(INSTALLATION_UUID, uuid);
+    } else {
+        if (differentDay(new Date(lastUpdatedDate), today)) {
+            let uuid = ctx.globalState.get(INSTALLATION_UUID);
+            if (uuid === undefined) {
+                console.log("missing uuid");
+                return;
+            }
```
Comment (DanCech) [inline src/telemetry.ts:30]:
```suggestion
            if (uuid === undefined) {
                uuid = uuidv4();
                ctx.globalState.update(INSTALLATION_UUID, uuid);
            }
```

File: src/telemetry.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        }
+    }
+}
+
+function differentDay(d1: Date, d2: Date) {
+    d1.getMonth() !== d2.getMonth() &&
+    d1.getFullYear() !== d2.getFullYear());
+    return d1.getDay() !== d2.getDay() &&
+           d1.getMonth() !== d2.getMonth() &&
+           d1.getFullYear() !== d2.getFullYear();
```
Comment (DanCech) [inline src/telemetry.ts:42]:
```suggestion
    return d1.getDate() !== d2.getDate() ||
           d1.getMonth() !== d2.getMonth() ||
           d1.getFullYear() !== d2.getFullYear();
```

File: src/telemetry.ts (hunk 1 above)
Comment (malcolmholmes) [reply src/telemetry.ts:30]: This state should never happen though. In which case, perhaps it doesn't matter setting a UUID vs logging an error. Hmm.

File: src/telemetry.ts (hunk 2 above)
Comment (malcolmholmes) [reply src/telemetry.ts:42]: oops, some logging snuck in. thx.

File: src/telemetry.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import {v4 as uuidv4} from 'uuid';
+import * as util from "./util";
+
+const LAST_UPDATED_DATE = "lastUpdatedDate";
+const INSTALLATION_DATE = "installDate";
+const INSTALLATION_UUID = "installUUID";
+const URL = "https://stats.grafana.org/vscode-usage-report";
+
+/*
+ * Sends a single anonymous telemetry call once per month, allowing tracking of
```
Comment (julienduchesne) [inline src/telemetry.ts:12]: Says once per month here, but the test is `differentDay` so it really is once a day, right?

**Review by julienduchesne — APPROVED** [review]
LGTM

File: src/telemetry.ts (hunk 3 above)
Comment (malcolmholmes) [reply src/telemetry.ts]: well spotted, fixed.

**Pull request merged**